MapboxVision/src/main/java/com/mapbox/vision/utils/observable/ObserverComposer.kt

```diff
@@ -0,0 +1,36 @@
+package com.mapbox.vision.utils.observable
+
+import java.lang.ref.WeakReference
+import java.util.concurrent.CopyOnWriteArraySet
+
+open class ObserverComposer<T> : Observable<T> {
+    private val set = CopyOnWriteArraySet<WeakReference<T>>()
```

Comment:
```suggestion
    private val weakReferenceSet = CopyOnWriteArraySet<WeakReference<T>>()
```

Reply:
I'd rename them just to `observers` or `weakObservers`, no need to know that it's a set IMO